app/controllers/users/sessions_controller.rb

```diff
@@ -1,2 +1,13 @@
 class Users::SessionsController < Devise::SessionsController
+  def new
+    super do |resource|
+      # When warden and devise threw an authentication error (invalid password)
```

Comment:
One alternative I can suggest:

https://stackoverflow.com/a/32655614 remarks that we could delete _all_ flash messages for new. I think that's better. What do you think? (Non-blocking.)

In fact, our Users::SessionsController could remove _all_ flash messages in general.